File: app/src/main/java/com/example/nazmuddinmavliwala/ewallet/base/views/BaseActivity.java
package com.example.nazmuddinmavliwala.ewallet.base.views;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import butterknife.ButterKnife;

/**
 * Created by nazmuddinmavliwala on 27/10/2017.
 */

public abstract class BaseActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(getLayout());
        ButterKnife.bind(this);
    }

    protected abstract int getLayout();

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
    }
}

File: app/src/main/java/com/chandilsachin/personal_finance/util/lifecycle/arch/BaseActivity.kt
package com.ace.diettracker.util.lifecycle.arch

import android.app.ProgressDialog
import android.os.Bundle
import android.os.PersistableBundle
import android.support.v7.app.AppCompatActivity
import com.ace.diettracker.util.SimpleProgressDialog

/**
 * Created by sachin on 27/6/17.
 */
open class BaseActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?, persistentState: PersistableBundle?) {
        super.onCreate(savedInstanceState, persistentState)

    }
}